File: AlgorithmsProject/src/in/algorithms/duplicatecharacters/DuplicateCharacterChecker.scala
package in.algorithms.duplicatecharacters

object DuplicateCharacterChecker {
  def checkIfContainsDuplicateCharacters(sentence : String) : Boolean = {
    val charBooleans : Array[Boolean] = Array.ofDim(256)
    for(i <- 0 to sentence.length() - 1) {
      val character = sentence(i)
      if(charBooleans(character)) return true
      else charBooleans(character) = true
    }
    return false
  }

  def main(args: Array[String]) {
   println( checkIfContainsDuplicateCharacters("Aabc"))
   println( checkIfContainsDuplicateCharacters("Vivek babu"))
   
  }

}

File: AlgorithmsProject/src/in/algorithms/duplicatecharacters/DuplicateCharacterChecker.java
package in.algorithms.duplicatecharacters;

import java.util.HashSet;
import java.util.Set;

public class DuplicateCharacterChecker {
    public static boolean checkIfContainsDuplicateCharacters(String str) {
        if (str == null) return false;
        Set<Character> seen = new HashSet<>();
        for (char c : str.toCharArray()) {
            if (!seen.add(c)) return true;
        }
        return false;
    }
}
